### PR TITLE
Simplify development setup

### PR DESCRIPTION
**Pull request by abraham**
## What type of PR is this? (check all applicable)

- [x] Refactor
- [ ] Feature
- [ ] Bug Fix
- [x] Documentation Update

## Description

Reduces the steps for setting up development (11 steps mac/linux and 13 steps windows to 7 steps for mac/linux) by using convention of the default database config, removing manual steps that `bin/setup` handles, and adding `foremen` to development gems group.

## Related Tickets & Documents

https://github.com/thepracticaldev/dev.to/pull/1162

## Mobile & Desktop Screenshots/Recordings (if there are UI changes)

n/a

## Added to documentation?

- [x] docs.dev.to
- [x] readme
- [ ] no documentation needed

## [optional] What gif best describes this PR or how it makes you feel?

working out

### CODE REVIEW

File: docs/installation/windows.md [hunk 1]
```diff
@@ -97,53 +97,28 @@ Pay attention to the username and password you setup during installation of Post
...
-1.  Fork dev.to repository, ie. https://github.com/thepracticaldev/dev.to/fork
-2.  Clone your forked repository, ie. `git clone https://github.com/<your-username>/dev.to.git`
-3.  Install bundler with `gem install bundler`
-4.  Install foreman gem with `gem install foreman`
-5.  Run `rbenv rehash` when foreman installation is done.
-6.  Install our ruby dependencies with `bundle install`
-    While installing, you might run into an error due to the pg gem requiring PostgreSQL libraries. If so, please run `sudo apt-get install libpq-dev` before retrying.
-7.  Setup your database
+1. Fork dev.to repository, ie. https://github.com/thepracticaldev/dev.to/fork
+1. Clone your forked repository, ie. `git clone https://github.com/<your-username>/dev.to.git`
```
Comment (rhymes) [inline docs/installation/windows.md]: the points became all 1s hehehe :D

File: docs/installation/windows.md [hunk 2]
```diff
@@ -97,53 +97,28 @@ Pay attention to the username and password you setup during installation of Post
...
-    username: # your Postgres username
-    password: # your Postgres password
-    ```
-
-    The database username and password also have to be manually added to test database in database.yml
-    Another possible way to configure database connection is to setup DATABASE_URL in application.yml as documented [here](https://docs.dev.to/additional-postgres-setup/#configuration)
-
-13. Run `bin/setup`
+1. Run `bin/setup`
+   - While installing, you might run into an error due to the pg gem requiring PostgreSQL libraries. If so, please run `sudo apt-get install libpq-dev` before retrying.
```
Comment (rhymes) [inline docs/installation/windows.md]: this should be moved to the linux guide

File: Gemfile [hunk 3]
```diff
@@ -122,8 +123,8 @@ group :development, :test do
   gem "capybara", "~> 3.13"
   gem "derailed", "~> 0.1"
   gem "erb_lint", "~> 0.0", require: false
-  gem "faker", git: "https://github.com/stympy/faker.git", branch: "master"
-  gem "fix-db-schema-conflicts", github: "thepracticaldev/fix-db-schema-conflicts", branch: "master"
+  gem "faker", github: "stympy/faker", ref: "e757835"
```
Comment (rhymes) [inline Gemfile]: since you submit a separate PR for faker and fix-db-schema-conflicts this should probably be removed from this PR

File: spec/factories/badges.rb [hunk 4]
```diff
@@ -4,7 +4,7 @@
   )
 
   factory :badge do
-    title { Faker::Overwatch.quote }
+    title { Faker::Games::Overwatch.quote }
```
Comment (rhymes) [inline spec/factories/badges.rb]: since you submit a separate PR for faker and fix-db-schema-conflicts this and the other related changes should probably be removed from this PR

File: docs/installation/windows.md (hunk 1 above)
Comment (kontrollanten) [reply docs/installation/windows.md]: Markdown converts all the 1s to accurate numeric values 💪

Comment (rhymes) [reply docs/installation/windows.md]: oooooooh didn't know that! thanks, you made my day!!!!

File: docs/installation/windows.md (hunk 2 above)
Comment (abraham) [reply docs/installation/windows.md]: Moved.

File: docs/installation/windows.md (hunk 1 above)
Comment (abraham) [reply docs/installation/windows.md]: It mostly works but you have to remember to indent anything in between the numbered lines.

File: spec/factories/badges.rb (hunk 4 above)
Comment (abraham) [reply spec/factories/badges.rb]: `foreman` and `faker` use incompatible versions of `thor` so this work has to be based off of #2142.

File: Gemfile (hunk 3 above)
Comment (abraham) [reply Gemfile]: `foreman` and `faker` use incompatible versions of `thor` so this work has to be based off of #2142.

Comment (abraham) [reply Gemfile]: Never mind. I was wrong and have removed the faker update from this PR.

Comment (rhymes) [reply Gemfile]: I just remembered that foreman shouldn't be part of the Gemfile: 

> Ruby users should take care not to install foreman in their project's Gemfile.

from https://github.com/ddollar/foreman#installation

Comment (abraham) [reply Gemfile]: Interesting. I was just looking back through the history of that warning and it seems to step from the problem of [dependency version conflicts](https://github.com/ddollar/foreman/pull/437#issuecomment-41110303). `foreman` is a development dependency so I think it should in fact be listed in the Gemfile. Dependency version conflicts is already a potential issue with all the other gems anyways.

Comment (abraham) [reply Gemfile]: I [proposed](https://github.com/ddollar/foreman/pull/728) an updated phrasing.

File: Gemfile [hunk 5]
```diff
@@ -110,6 +110,7 @@ group :development do
   gem "bullet", "~> 5.9"
   gem "bundler-audit", "~> 0.6"
   gem "derailed_benchmarks", "~> 1.3"
+  gem "foreman", github: "andrewmcodes/foreman", ref: "b64e401", require: false # Switch to stable release when Thor dependecy is updated to 0.20+
```
Comment (maestromac) [inline Gemfile]: Will this ref still work if the branch which it belongs to is removed?

File: .travis.yml [hunk 6]
```diff
@@ -14,6 +14,7 @@ env:
   global:
     - RAILS_ENV=test
     - CC_TEST_REPORTER_ID=f39e060a8b1a558ebd8aff75d5b9760bf1ae98f3f85d628ae28814f3c66438cd
+    - DATABASE_URL=postgres://postgres@localhost/
```
Comment (maestromac) [inline .travis.yml]: Oh, I wish Travis doc stated this. It's so much more intuitive to do it this way.

File: Gemfile (hunk 5 above)
Comment (abraham) [reply Gemfile]: If the branch is deleted I think this would start failing. If you fork `andrew/foreman` under `thepracticledev` I'll update the repo until https://github.com/ddollar/foreman/pull/727 is merged.

File: .travis.yml (hunk 6 above)
Comment (abraham) [reply .travis.yml]: I've never been a super big fan of Travis docs.

File: Gemfile (hunk 5 above)
Comment (maestromac) [reply Gemfile]: Here you go
```suggestion
  gem "foreman", github: "thepracticaldev/foreman", ref: "b64e401", require: false # Switch to stable release when Thor dependecy is updated to 0.20+
```

Comment (abraham) [reply Gemfile]: @maestromac The fork doesn't have the `update-thor` branch so you'll need to copy that too.

Comment (maestromac) [reply Gemfile]: That won't be necessary since the ref match. I've tested it locally.

Comment (abraham) [reply Gemfile]: Cool!

**Comment by abraham** [comment]
Resolved `Gemfile.lock` conflict.

File: Gemfile (hunk 5 above)
Comment (abraham) [reply Gemfile]: Suggested changes probably shouldn't be used for `Gemfile` changes as the `Gemfile.lock` will be outdated. I've updated this branch's lock file but I missed it initially.

**Comment by lightalloy** [comment]
@maestromac If this way is more convenient to the community, I'm ok with the changes.
